pennylane/ops/op_math/sum.py

```diff
@@ -298,6 +285,36 @@ def simplify(self, cutoff=1.0e-12) -> "Sum":  # pylint: disable=arguments-differ
             return Sum(*new_summands) if len(new_summands) > 1 else new_summands[0]
         return qml.s_prod(0, qml.Identity(self.wires))
 
+    def terms(self):
+        r"""Representation of the operator as a linear combination of other operators.
+
+        .. math:: O = \sum_i c_i O_i
+
+        A ``TermsUndefinedError`` is raised if no representation by terms is defined.
+
+        Returns:
+            tuple[list[tensor_like or float], list[.Operation]]: list of coefficients :math:`c_i`
+            and list of operations :math:`O_i`
+        """
+        # try using pauli_rep:
+        if pr := self.pauli_rep:
+            pr.simplify()
```

Comment:
Not entirely sure we actually need this line, because it basically only eliminates terms with really small coefficients (less than `1e-8`).

Reply:
yep good point! Frankly I copy-pasted this from the simplify method